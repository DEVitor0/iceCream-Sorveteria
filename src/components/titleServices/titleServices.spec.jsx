import { render, screen } from '@testing-library/react';
import TitleServices from './index';

describe('TitleServices Component', () => {
  it('renders the TitleServices with text correctly', () => {
    render(<TitleServices text="Test Title" />);
    expect(screen.getByText('Test Title')).toBeInTheDocument();
  });

  it('applies "looking-services" class when id is provided', () => {
    render(<TitleServices id="test-id" text="Test Title" />);
    expect(screen.getByText('Test Title')).toHaveClass('looking-services');
  });

  it('does not apply "looking-services" class when id is not provided', () => {
    render(<TitleServices text="Test Title" />);
    expect(screen.getByText('Test Title')).not.toHaveClass('looking-services');
  });

  it('renders with the correct id attribute', () => {
    render(<TitleServices id="test-id" text="Test Title" />);
    expect(screen.getByText('Test Title')).toHaveAttribute('id', 'test-id');
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(<TitleServices text="Test Title" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
