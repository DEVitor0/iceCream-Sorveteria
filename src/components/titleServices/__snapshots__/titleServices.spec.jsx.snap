// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleServices Component matches the snapshot 1`] = `
<DocumentFragment>
  <h3
    class="title-services "
  >
    Test Title
  </h3>
</DocumentFragment>
`;
